saleor/payment/utils.py

```diff
@@ -0,0 +1,180 @@
+            type=OrderEvents.EMAIL_SENT.value,
+            parameters={
+                'email': order.get_user_current_email(),
+                'email_type': OrderEventsEmails.PAYMENT.value})
+    try:
+        analytics.report_order(order.tracking_client_id, order)
+    except Exception:
+        # Analytics failing should not abort the checkout flow
+        logger.exception('Recording order in analytics failed')
+
+
+def validate_payment(view):
+    """Decorate a view to check if payment is authorized, so any actions
+    can be performed on it.
+    """
+
+    @wraps(view)
+    def func(payment: Payment, *args, **kwargs):
+        if not payment.is_active:
+            raise PaymentError('This payment is no longer active.')
+        return view(payment, *args, **kwargs)
+    return func
+
+
+def create_payment(**payment_data):
+    payment, _ = Payment.objects.get_or_create(**payment_data)
+    return payment
+
+
+def create_transaction(
+        payment: Payment,
+        token: str,
+        transaction_type: str,
+        is_success: bool,
+        amount: Decimal,
+        currency: str,
+        gateway_response: Optional[Dict] = None) -> Transaction:
+    if not gateway_response:
+        gateway_response = {}
+    txn, _ = Transaction.objects.get_or_create(
+        payment=payment,
+        token=token,
+        transaction_type=transaction_type,
+        is_success=is_success,
+        amount=amount,
+        currency=currency,
+        gateway_response=gateway_response)
+    return txn
+
+
+def gateway_get_transaction_token(provider_name: str):
+    # FIXME Add tests
+    provider, provider_params = get_provider(provider_name)
+    return provider.get_transaction_token(**provider_params)
+
+
+@validate_payment
+def gateway_authorize(payment: Payment, transaction_token: str) -> Transaction:
+    if not payment.charge_status == ChargeStatus.NOT_CHARGED:
+        raise PaymentError('Charged transactions cannot be authorized again.')
+
+    provider, provider_params = get_provider(payment.variant)
+    with transaction.atomic():
+        txn, error = provider.authorize(
+            payment, transaction_token, **provider_params)
+    # FIXME Create an order event ?
+    if not txn.is_success:
+        raise PaymentError(error)
+    return txn
+
+
+@validate_payment
+def gateway_capture(payment: Payment, amount: Decimal) -> Transaction:
+    if amount <= 0:
+        raise PaymentError('Amount should be a positive number.')
+    if payment.charge_status not in {ChargeStatus.CHARGED,
+                                     ChargeStatus.NOT_CHARGED}:
+        raise PaymentError('This payment cannot be captured.')
+    if amount > payment.total or amount > (
+            payment.total - payment.captured_amount):
+        raise PaymentError('Unable to capture more than authorized amount.')
+
+    provider, provider_params = get_provider(payment.variant)
+    with transaction.atomic():
+        txn, error = provider.capture(
```

Comment:
In dashboard, if the provider doesn't support manual capturing, it now crashes when we try to capture. `django-payments` was checking whether manual capturing was supported or not.

Reply:
It's fixed on my local instance